### ISSUE TITLE
why i run train.py is ok,but run train_bottleneck is wrong:loss:nan,val_loss:nan

### ISSUE THREAD

**Issue by AryaCao** [qqwweee/keras-yolo3#305]
run the train_bottleneck :
 start like this:
![2018-12-20 01-40-49](https://user-images.githubusercontent.com/34637295/50212414-fb348d80-03b5-11e9-89a0-7a7ca59c7230.png)

later change to this:

![2018-12-20 01-40-59](https://user-images.githubusercontent.com/34637295/50212487-21f2c400-03b6-11e9-9d53-922522ccb1cb.png)

can someone tell me why?

**Comment by AryaCao** [comment]
It seems to be that the parameters:steps of " bottleneck_model.predict_generator" is wrong?

**Comment by Fingerl** [comment]
hello，can you tell me the different between the train.py and the train_bottleneck.py?Which one should I use?

**Comment by AryaCao** [comment]
hello , I think that the two file has same function. 
The 'train_bottlenck.py' can speed up the training process in cpu or gpu, because  it will compute the bottleneck features of the frozen model first(saved in the .npy files) and then load the features to  train the last layers. 
I think that you can just use the train.py, if you have a gpu (good compute capability ).

**Comment by Fingerl** [comment]
thank you

**Comment by matt-deboer** [comment]
I was able to fix this by adding the following lines around `383` of `yolo3/model.py`:
_(right below the existing `K.switch...` statement)_
```
# filter out inf/nan values encountered here:
raw_true_wh = K.switch(tf.is_inf(raw_true_wh), raw_true_wh, K.zeros_like(raw_true_wh)) 
raw_true_wh = K.switch(tf.is_nan(raw_true_wh), raw_true_wh, K.zeros_like(raw_true_wh))
```
This is a bit of a hack; there may be some data-related causes for those inf/nan values to occur, but I wasn't able to find them in my data set; then again, it can't hurt to be less sensitive to bogus data, right?

**Comment by CherishineNi** [comment]
@AryaCao 请问如何使用gpu运行train.py文件呢？需要额外的指令代码吗？

**Comment by monicatao** [comment]
I have the same problem. How did you solve it?
